### PR TITLE
Add separate EAN validator,

### PR DESCRIPTION
**Pull request by camillobruni**
In prepration for new Swiss SSN / AHV Field

### CODE REVIEW

File: localflavor/generic/validators.py [hunk 1]
```diff
@@ -209,3 +212,23 @@ def __call__(self, value):
...
+
+class EANValidator(object):
+    """
+    A generic validator for EAN like codes with the last digit being the checksum.
+
+    http://en.wikipedia.org/wiki/International_Article_Number_(EAN)
+    """
+
+    def __init__(self, strip_nondigits=False, error_message=None):
+        self.error_message = error_message
```
Comment (claudep) [inline localflavor/generic/validators.py]: Could we mimic the error message implementation of Django core validators? That is a default `message = _(...)` on the class, a `message` keyword argument and the `if message is not None:  self.message = message` sequence in `__init__`.

File: localflavor/generic/validators.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
+import re
```
Comment (claudep) [inline localflavor/generic/validators.py]: nitpick: blank line before, not after `import re`

File: tests/test_generic.py [hunk 3]
```diff
@@ -337,3 +337,63 @@ def test_bic_model_field(self):
...
+            None,
+        ]
+        error_message = 'Invalid EAN Code'
+        invalid = {
+            '400.6381.3339.31',
+            '4006381333930',
+            '',
+            '0',
+            'DÉUTDEFF',
+        }
```
Comment (claudep) [inline tests/test_generic.py]: This set constructor syntax is not supported on Python 2.6 and localflavor is still supporting Python 2.6.

File: tests/test_generic.py [hunk 4]
```diff
@@ -337,3 +337,63 @@ def test_bic_model_field(self):
...
+        ]
+        error_message = 'Invalid EAN Code'
+        invalid = {
+            '4006381333930',
+            '400-63-813-339-30',
+            '400 63 813 339 30',
+            '',
+            '0',
+            'DÉUTDEFF',
+        }
```
Comment (claudep) [inline tests/test_generic.py]: Same as above.

File: tests/test_generic.py (hunk 3 above)
Comment (camillobruni) [inline tests/test_generic.py]: should've been a list, thanks

**Comment by claudep** [comment]
Tests are now failing with Python 3, see Travis CI details.

**Pull request merged**